lib/Basic/LangOptions.cpp

```diff
@@ -392,6 +393,10 @@ void LangOptions::setHasAtomicBitWidth(llvm::Triple triple) {
     setMaxAtomicBitWidth(128);
     break;
 
+  case llvm::Triple::ArchType::avr:
+    setMaxAtomicBitWidth(32);
```

Comment:
I think this deserves some comment/explanation, because IIUC this is sort of a lie? I think that all the other CPU cases declare what is the maximum *lock-free hw supported* atomic bit width, and on AVR that's really... nothing? And software-assisted atomics (via a lock or by disabling interrupts) can be done on any architecture and for any bitwidth, so that's not really that meaningful either.

Reply:
Alright. I see your point. I'll remove this from this PR. We are skipping the build of the Synchronisation library on AVR anyway, which I think is where it was needed.